### PR TITLE
First steps to allowing Linux to use EGL and OpenGLES

### PR DESCRIPTION
**Pull request by prashantgupta24**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8869)

### CODE REVIEW

**Comment by KiChjang** [comment]
Just a tip here - you can use `./mach test-tidy` to check whether your commits fit our styling instead of using Travis-CI to check.

**Comment by glennw** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[ports/glutin/window.rs, line 142 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8869#-K4wTK0vWRnLXPNOoLzH:-K4wTK0vWRnLXPNOoLzI:-1741190877) ([raw file](https://github.com/servo/servo/blob/1e2a14525a3f22d3a7e90ed61709c156cc771611/ports/glutin/window.rs#L142)):*
This implementation of the function (that selects between GL / ES2) should only exist on Linux. On Android, we can only support ES2, so the old function that always returns ES should continue to exist on Android.

---

*[ports/glutin/window.rs, line 144 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8869#-K4wTXdCdQIwRXzL97Sk:-K4wTXdCdQIwRXzL97Sl:400090728) ([raw file](https://github.com/servo/servo/blob/1e2a14525a3f22d3a7e90ed61709c156cc771611/ports/glutin/window.rs#L144)):*
Indentation is wrong here - switch tabs to spaces.

---

*[ports/glutin/window.rs, line 643 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8869#-K4wTc67TWtIf5LE74J9:-K4wTc67TWtIf5LE74JA:987241256) ([raw file](https://github.com/servo/servo/blob/1e2a14525a3f22d3a7e90ed61709c156cc771611/ports/glutin/window.rs#L643)):*
Indentation is wrong here - tabs to spaces.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8869)*

**Comment by glennw** [comment]
Just a few minor changes left - nearly there! :)

File: components/util/opts.rs [hunk 1]
```diff
@@ -188,6 +188,9 @@ pub struct Opts {
 
     /// Do not use native titlebar
     pub no_native_titlebar: bool,
+
+    //This variable is of the type enum Render API defined below, used for selecting graphic options dynamically
```
Comment (glennw) [inline components/util/opts.rs]: This should be a doc comment (3 slashes) - something like "Select GLX or EGL backend (Linux only)"

File: components/util/opts.rs [hunk 2]
```diff
@@ -488,6 +499,7 @@ pub fn default_opts() -> Opts {
         convert_mouse_to_touch: false,
         exit_after_load: false,
         no_native_titlebar: false,
+	graphics_select: DEFAULT_GRAPHICS,
```
Comment (glennw) [inline components/util/opts.rs]: nit: identation still seems slightly off here

File: ports/glutin/window.rs [hunk 3]
```diff
@@ -138,15 +139,19 @@ impl Window {
         }
     }
 
-    #[cfg(not(target_os = "android"))]
+#[cfg(target_os = "linux")]
     fn gl_version() -> GlRequest {
-        GlRequest::Specific(Api::OpenGl, (2, 1))
+        match opts::get().graphics_select {
```
Comment (glennw) [inline ports/glutin/window.rs]: Indentation still seems wrong here (should be 4 spaces)

**Comment by nox** [comment]
This needs to be rebased.

**Comment by frewsxcv** [comment]
@prashantgupta24 need any help rebasing this?

**Comment by nox** [comment]
I did this: https://github.com/servo/servo/compare/master...nox:refactor_gle

Is that the way to go now that we have webrender?

**Comment by jdm** [comment]
Rebased in #9781.

**Pull request closed without merging**